src/lib/peer.js

```diff
@@ -79,9 +91,11 @@ function Peer(config) {
 
     function router(envelope) {
         var nextHop = self.fingerTable.bestCandidate(envelope.dstId);
-        log('nextHop:', nextHop, envelope);
-        if (nextHop === self.peerId.toHex()) {
+        console.log('nextHop:', nextHop, envelope);
+        if (nextHop === self.peerId.toHex() && envelope.dstId === self.peerId.toHex()) {
```

Comment:
In that scenario, if a msg has dstID: peer4, since there is no way for a given node to have a snapshot of the entire network (nor it is desired), what happens is that the message is going to traverse the ring and when it arrives to peer1, peer1 will see "ah, this message arrived to me, and it is Id 'before' than mine, so it is for me". It might sound not efficient, since after all the message was destined to peer1 after all, but since it was him sending it, he will only know that there were no peers available before him when the message arrives.